test/provider_os_memory_multiple_numa_nodes.cpp

```diff
@@ -88,13 +95,21 @@ struct testNuma : testing::Test {
         ASSERT_NE(os_memory_provider, nullptr);
     }
 
-    struct bitmask *retrieve_nodemask(void *addr) {
+    std::pair<int, bitmask *> retrieve_nodemask(void *addr) {
         struct bitmask *retrieved_nodemask = numa_allocate_nodemask();
-        UT_ASSERTne(nodemask, nullptr);
+
+        if (nodemask == nullptr) {
+            return std::make_pair(-1, nodemask);
+        }
+
         int ret = get_mempolicy(nullptr, retrieved_nodemask->maskp,
                                 nodemask->size, addr, MPOL_F_ADDR);
-        UT_ASSERTeq(ret, 0);
-        return retrieved_nodemask;
+
```

Comment:
why can't we use asserts here and just check it like in your other PRs (`IS_SKIPPED_OR_FAILED`)?

Reply:
Changed as you wished, the output message is more precise with your suggestion